pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java

```diff
@@ -130,6 +133,15 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
         tableSubtypeSize.estimatedSizeInBytes,
         tableSubtypeSize.reportedSizeInBytes);
 
+    // Only emit the real percentage of storage quota usage by lead controller, otherwise emit 0L.
+    if (_pinotHelixResourceManager.isLeader()) {
+      long existingStorageQuotaUtilization = tableSubtypeSize.estimatedSizeInBytes / allowedStorageBytes * 100;
```

Comment:
We should be careful about allowedStorageBytes == 0. There is a < 0 check for that earlier in the code, but not for == 0. Please add a check to avoid div by zero errors.